src/app/singup/singup.component.ts

```diff
@@ -53,10 +53,6 @@ export class SingupComponent implements OnInit {
   }
 
   addUser() {
-    if (this.userForm.invalid) {
-      return;
-    } else {
-      this.userService.addUser(this.user);
-    }
+    this.userService.addUser(this.user);
```

Comment:
why did you remove that validation ?

Reply:
because, button is enable when all fields are correct
if any field isn't invalid(length, required, email etc) button comes disabled

Reply:
In the following comments, I will leave more informative comments about the changes

Reply:
you can open devtools in browser and remove "disabled" attribute manually. In such case you won't have any validation enabled and frontend will send a bad request to backend.

Reply:
done